### PR TITLE
made Create and FindById functions in householdModel file

### PR DESCRIPTION
**Pull request by samplesn08**
# Description

Summary of your changes
- made create function for households
- made findById function for households
## What work was done?

Details of your changes
-created the 'create' and 'findById' functions in the householdModel file
## Why was this work done?

Why?
-in order to give the user the ability to create a new household in the database
## Type of change

- New feature (non-breaking change which adds functionality)

## Change Status

- Completed, ready to review and merge

## Checklist

- [ x] My code follows the style guidelines of this project
- [x ] I have performed a self-review of my own code
- [x ] My changes generate no new warnings
- [x ] There are no merge conflicts

### CODE REVIEW

**Review by blackcatwizard — APPROVED** [review]
this looks good!

**Review by Bchoi562 — APPROVED** [review]

File: api/household/householdModel.js [hunk 1]
```diff
@@ -0,0 +1,44 @@
+const knex = require('../../data/db-config');
+
+const create = async (households) => {
+  let newHouseholdId;
+  try {
+    await knex.transaction(async (trx) => {
```
Comment (jaypdl) [inline api/household/householdModel.js:6]: It doesn't harm anything in this case, but a transaction isn't necessary here. 

It's only needed when you need to do insertions into several different tables at once.  

How it works, is that if something goes wrong when inserting into one of the tables, all the changes so far get rolled back. In that way, you don't have a situation where you have random pieces of data added to your tables.

**Review by jaypdl — APPROVED** [review]
Great job with adding in an endpoint!

File: api/household/householdModel.js (hunk 1 above)
Comment (samplesn08) [reply api/household/householdModel.js:6]: Good to know!  Thank you

File: api/household/householdModel.js [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+  } catch (err) {
+    throw new Error(err);
+  }
+};
+
+const findById = async (id) => {
+  return await knex('households')
+    .where({ 'households.household_id': id })
+    .leftJoin('locations', {
+      'households.household_id': 'locations.household_id',
```
[the remark this answers is not in the document]
Comment (samplesn08) [reply api/household/householdModel.js:28]: Ah!  You're correct!  Should be households.location_id and locations.location_id.  I'll fix it.  Good catch!

**Review by DomenicScarcella — APPROVED** [review]
Looks good!

**Pull request merged**